sdk/src/types/block/semantic.rs

```diff
@@ -209,294 +210,279 @@ impl<'a> ValidationContext<'a> {
+                        && !address.has_capability(AddressCapabilityFlag::OutputsWithTimelock)
+                    {
+                        // TODO: add a variant https://github.com/iotaledger/iota-sdk/issues/1430
+                        return Ok(Some(TransactionFailureReason::SemanticValidationFailed));
+                    }
+
+                    if unlock_conditions.expiration().is_some()
+                        && !address.has_capability(AddressCapabilityFlag::OutputsWithExpiration)
+                    {
+                        // TODO: add a variant https://github.com/iotaledger/iota-sdk/issues/1430
+                        return Ok(Some(TransactionFailureReason::SemanticValidationFailed));
+                    }
+
+                    if unlock_conditions.storage_deposit_return().is_some()
+                        && !address.has_capability(AddressCapabilityFlag::OutputsWithStorageDepositReturn)
+                    {
+                        // TODO: add a variant https://github.com/iotaledger/iota-sdk/issues/1430
+                        return Ok(Some(TransactionFailureReason::SemanticValidationFailed));
+                    }
+
+                    if match &created_output {
+                        Output::Account(_) => !address.has_capability(AddressCapabilityFlag::AccountOutputs),
+                        Output::Nft(_) => !address.has_capability(AddressCapabilityFlag::NftOutputs),
+                        Output::Delegation(_) => !address.has_capability(AddressCapabilityFlag::DelegationOutputs),
+                        _ => false,
+                    } {
+                        // TODO: add a variant https://github.com/iotaledger/iota-sdk/issues/1430
+                        return Ok(Some(TransactionFailureReason::SemanticValidationFailed));
+                    }
                 }
+            }
 
-                if created_output.mana() > 0 && !address.has_capability(AddressCapabilityFlag::OutputsWithMana) {
-                    // TODO: add a variant https://github.com/iotaledger/iota-sdk/issues/1430
-                    return Ok(Some(TransactionFailureReason::SemanticValidationFailed));
+            if let Some(sender) = features.and_then(|f| f.sender()) {
+                if !self.unlocked_addresses.contains(sender.address()) {
+                    return Ok(Some(TransactionFailureReason::SenderNotUnlocked));
                 }
+            }
 
-                if unlock_conditions.timelock().is_some()
-                    && !address.has_capability(AddressCapabilityFlag::OutputsWithTimelock)
-                {
-                    // TODO: add a variant https://github.com/iotaledger/iota-sdk/issues/1430
-                    return Ok(Some(TransactionFailureReason::SemanticValidationFailed));
-                }
+            self.output_amount = self
+                .output_amount
+                .checked_add(amount)
+                .ok_or(Error::CreatedAmountOverflow)?;
```

Comment:
would be kinda neat to rename this error to `CreatedCoinOverflow`

Reply:
I would keep amount as it's the field name

Reply:
okay